lib/improver/tests/spotdata/spotdata/test_read_input.py

```diff
@@ -0,0 +1,360 @@
+                             units='degrees', var_name='longitude')
+
+        # Use time of 2017-02-17 06:00:00, 07:00:00
+        time = DimCoord(
+            [1487311200, 1487314800], standard_name='time',
+            units=cf_units.Unit('seconds since 1970-01-01 00:00:00',
+                                calendar='gregorian'), var_name='time')
+
+        time_dt = dt(2017, 2, 17, 6, 0)
+        time_extract = Constraint(time=PartialDateTime(
+            time_dt.year, time_dt.month, time_dt.day, time_dt.hour))
+
+        cube = Cube(data,
+                    long_name="air_temperature",
+                    dim_coords_and_dims=[(time, 0),
+                                         (latitude, 1),
+                                         (longitude, 2)],
+                    units="K")
+        cube2 = cube.copy()
+
+        orography = Cube(np.ones((20, 20)),
+                         long_name="surface_altitude",
+                         dim_coords_and_dims=[(latitude, 0),
+                                              (longitude, 1)],
+                         units="m")
+
+        land = orography.copy()
+        land.rename('land_binary_mask')
+        land.data = land.data + 1
+
+        ancillary_data = {}
+        ancillary_data.update({'orography': orography})
+        ancillary_data.update({'land_mask': land})
+
+        # Copies of cube simply renamed to be read as additional data.
+        temperature_on_height_levels = cube.copy()
+        temperature_on_height_levels.rename('temperature_on_height_levels')
+        pressure_on_height_levels = cube.copy()
+        pressure_on_height_levels.rename('pressure_on_height_levels')
+        surface_pressure = cube.copy()
+        surface_pressure.rename('surface_pressure')
+
+        # Build reference copy of additional_data dictionary.
+        with iris.FUTURE.context(cell_datetime_objects=True):
+            additional_data = {
+                'temperature_on_height_levels': CubeList(
+                    [temperature_on_height_levels]),
+                'pressure_on_height_levels': CubeList([
+                    pressure_on_height_levels]),
+                'surface_pressure': CubeList([surface_pressure])
+                }
+
+        self.data_directory = mkdtemp()
+
+        self.cube_path = (self.data_directory +
+                          '/01-temperature_at_screen_level.nc')
+        self.cube_path2 = (self.data_directory +
+                           '/02-temperature_at_screen_level.nc')
+        orography_path = self.data_directory + '/orography.nc'
+        land_path = self.data_directory + '/land_mask.nc'
+        ad_path_temperature = (self.data_directory +
+                               '/temperature_on_height_levels.nc')
+        ad_path_pressure = (self.data_directory +
+                            '/pressure_on_height_levels.nc')
+        ad_path_s_pressure = self.data_directory + '/surface_pressure.nc'
+
+        iris.save(cube, self.cube_path)
```

Comment:
For StaGE we put some small data files in the repo for testing reading, but I think this is better.